jni/src/knn_extension/faiss/utils/BitSet.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <assert.h>
+#include "BitSet.h"
+
+FixedBitSet::FixedBitSet(const int* int_array, const int length){
+    assert(int_array && "int_array should not be null");
+    int maxValue = int_array[length - 1];
+    this->numBits = (maxValue >> 6) + 1; // div by 64
+    this->bitmap = new uint64_t[this->numBits]();
+    for(int i = 0 ; i < length ; i ++) {
+        int value = int_array[i];
+        int bitsetArrayIndex = value >> 6;
+        this->bitmap[bitsetArrayIndex] |= 1ULL << value; // Equivalent of 1ULL << (value % 64)
+    }
+}
+
+idx_t FixedBitSet::next_set_bit(idx_t index) {
+    idx_t i = index >> 6; // div by 64
+    uint64_t word = this->bitmap[i] >> index; // Equivalent of bitmap[i] >> (index % 64)
```

Comment:
same as above.

```suggestion
    uint64_t word = this->bitmap[i] >> (index & 64); // Equivalent of bitmap[i] >> (index % 64)
```

Reply:
Also, bitmap[i] is your word. why we are doing `this->bitmap[i] >> index` to get the word?

Reply:
It is removing all right side bits from current index so that we can find the next bit by counting the trailing zero bit.